src/executor/execute.rs

```diff
@@ -22,12 +22,23 @@ pub(crate) enum ExecResultHolder {
     /// Holds the result of executing an operation on the Mock database
     #[cfg(feature = "mock")]
     Mock(crate::MockExecResult),
+    Disconnected,
```

Comment:
While I agree on adding Disconnected state to InnerConnection, I don't think it makes sense to add on ExecResultHolder. What problem does it solve here?

Reply:
Not really. I originally want to have a placeholder to avoid the use of

```rs
#[allow(unreachable_patterns)]
_ => unreachable!(),
```